data/model/namespacequota.py

```diff
@@ -0,0 +1,287 @@
+import json
+
+from peewee import fn, JOIN
+
+from data.database import (
+    Repository,
+    QuotaLimits,
+    UserOrganizationQuota,
+    QuotaType,
+    Manifest,
+    get_epoch_timestamp_ms,
+    Tag,
+    RepositorySize,
+)
+from data.model import (
+    InvalidOrganizationException,
+    _basequery,
+    organization,
+    user,
+    InvalidUsernameException,
+    repository,
+    notification,
+)
+
+
+def verify_namespace_quota(namespace_name, repository_ref):
+    repository.get_repository_size_and_cache(repository_ref._db_id)
+    namespace_size = get_namespace_size(namespace_name)
+    return check_limits(namespace_name, namespace_size)
+
+
+def verify_namespace_quota_force_cache(namespace_name, repository_ref):
+    force_cache_repo_size(repository_ref)
+    namespace_size = get_namespace_size(namespace_name)
+    return check_limits(namespace_name, namespace_size)
+
+
+def verify_namespace_quota_during_upload(namespace_name, repository_ref):
```

Comment:
Why do you count BlobUploads size only during upload?

Reply:
We want to ensure that if they are doing a large push that as the blobs add up during upload we can exit prior to a full upload.